src/index.js

```diff
@@ -130,41 +136,81 @@ class Serverless_WSO2_APIM {
   }
 
   async validateConfig() {
-    // this.serverless.cli.log(pluginNameSuffix + "Validating configuration..");
-    const wso2APIM = this.serverless.service.custom.wso2apim;
-    if (
-      wso2APIM.host == undefined ||
-      wso2APIM.port == undefined ||
-      wso2APIM.user == undefined ||
-      wso2APIM.pass == undefined ||
-      wso2APIM.apidefs == undefined
-    ) {
+    try {
+      // this.serverless.cli.log(pluginNameSuffix + "Validating configuration..");
+      const wso2APIM = this.serverless.service.custom.wso2apim;
+
+      // Key value checks, with corresponding error messages
+      const conditionsArray = [
+        ((wso2APIM.enabled) && ([true, false].includes(wso2APIM.enabled))),
+        ((wso2APIM.host) && (wso2APIM.host.length > 0)),
+        ((wso2APIM.port) && (wso2APIM.port > 0)),
+        ((wso2APIM.user) && (wso2APIM.user.length > 0)),
+        ((wso2APIM.pass) && (wso2APIM.pass.length > 0)),
+        ((wso2APIM.gatewayEnv) && (wso2APIM.gatewayEnv.length > 0)),
+        (wso2APIM.apidefs.length > 0)
+      ];
+      const messagesArray = [
```

Comment:
I think there's a need for schema validation here. Joi (https://www.npmjs.com/package/joi) can do this, you can use it to validate the configuration you're getting in wsoAPIM against a schema.

Reply:
Ok thats great to know. Let me investigate it.

Reply:
As a part of a different PR, I was thinking of using jsonschema & ajv to get the schema validation achieved. What do you think?